openedx/core/djangoapps/credit/email_utils.py

```diff
@@ -0,0 +1,138 @@
+"""
+This file contains utility functions which will responsible for sending emails.
+"""
+
+import os
+
+import logging
+import pynliner
+import urlparse
+import uuid
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.contrib.staticfiles import finders
+from django.core.mail import EmailMessage
+from django.core.urlresolvers import reverse
+from django.utils.translation import ugettext as _
+
+from opaque_keys.edx.keys import CourseKey
+
+from edxmako.shortcuts import render_to_string
+from email.mime.image import MIMEImage
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from eventtracking import tracker
+from microsite_configuration import microsite
+from openedx.core.djangoapps.user_api.accounts.api import get_account_settings
+from xmodule.modulestore.django import modulestore
+
+
+log = logging.getLogger(__name__)
+
+
+def send_credit_notifications(username, course_key, email_type="credit_eligibility"):
+    """Sends email notification to user on different phases during credit
+    course e.g., credit eligibility, credit payment etc.
+    """
+    try:
+        user = User.objects.get(username=username)
+    except User.DoesNotExist:
+        log.error('No user with %s exist', username)
+        return
+
+    account_settings = get_account_settings(user)
```

Comment:
I noticed that the number of queries went up a lot, most likely because of this call.  But all you're getting from this is the user's full name.  I think you could do this in one query by using `select_related` for the `UserProfile` object and getting the full name from that.